### PR TITLE
[Doc] Improve AutocompleteArrayInput doc

### PR DESCRIPTION
**Pull request by fzaninotto**

### CODE REVIEW

File: docs/AutocompleteArrayInput.md [hunk 1]
```diff
@@ -10,125 +10,396 @@ It renders using MUI [Autocomplete](https://mui.com/components/autocomplete/).
...
+const choices = [
+    { id: 'admin', label: 'myroot.roles.admin' },
+    { id: 'u001', label: 'myroot.roles.u001' },
+    { id: 'u002', label: 'myroot.roles.u002' },
+    { id: 'u003', label: 'myroot.roles.u003' },
+];
 ```
 
-You should use the following syntax:
+You can opt-out of this translation by setting [the `translateChoice` prop](#translatechoice) to `false`.
```
Comment (WiXSL) [inline docs/AutocompleteArrayInput.md:108]: #translatechoice is missing

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/AutocompleteArrayInput.md (hunk 1 above)
Comment (fzaninotto) [reply docs/AutocompleteArrayInput.md:110]: Right, fixed

File: docs/AutocompleteArrayInput.md [hunk 2]
```diff
@@ -10,125 +10,418 @@ It renders using MUI [Autocomplete](https://mui.com/components/autocomplete/).
...
 The choices are translated by default, so you can use translation identifiers as choices:
 
 ```jsx
 const choices = [
-   { id: 'M', name: 'myroot.gender.male' },
-   { id: 'F', name: 'myroot.gender.female' },
+    { id: 'admin', label: 'myroot.roles.admin' },
+    { id: 'u001', label: 'myroot.roles.u001' },
+    { id: 'u002', label: 'myroot.roles.u002' },
+    { id: 'u003', label: 'myroot.roles.u003' },
```
Comment (slax57) [inline docs/AutocompleteArrayInput.md:411]:
```suggestion
    { id: 'admin', name: 'myroot.roles.admin' },
    { id: 'u001', name: 'myroot.roles.u001' },
    { id: 'u002', name: 'myroot.roles.u002' },
    { id: 'u003', name: 'myroot.roles.u003' },
```

File: docs/AutocompleteArrayInput.md [hunk 3]
```diff
@@ -10,125 +10,418 @@ It renders using MUI [Autocomplete](https://mui.com/components/autocomplete/).
...
 ];
 ```
 
-However, in some cases (e.g. inside a `<ReferenceInput>`), you may not want the choice to be translated. In that case, set the `translateChoice` prop to `false`.
+However, in some cases (e.g. inside a `<ReferenceArrayInput>`), you may not want the choice to be translated.
+In that case, set the `translateChoice` prop to `false`.
 
 ```jsx
-<AutocompleteArrayInput source="gender" choices={choices} translateChoice={false}/>
+<AutocompleteArrayInput source="tags" choices={choices} translateChoice={false}/>
```
Comment (slax57) [inline docs/AutocompleteArrayInput.md:419]:
```suggestion
<AutocompleteArrayInput source="roles" choices={choices} translateChoice={false}/>
```

**Review by slax57 — CHANGES_REQUESTED** [review]

File: docs/AutocompleteArrayInput.md [hunk 4]
```diff
@@ -10,125 +10,418 @@ It renders using MUI [Autocomplete](https://mui.com/components/autocomplete/).
...
+| `choices`                  | Required | `Object[]`            | -                        | List of choices                                                   |
+| `create`                   | Optional | `Element`             | `-`                      | A React Element to render when users want to create a new choice                                                                       |
+| `createLabel`              | Optional | `string`              | `ra.action. create`       | The label for the menu item allowing users to create a new choice. Used when the filter is empty                                       |
+| `createItemLabel`          | Optional | `string`              | `ra.action .create_item` | The label for the menu item allowing users to create a new choice. Used when the filter is not empty                                                    |
+| `debounce`                 | Optional | `number`              | `250`                    | The delay to wait before calling the setFilter function injected when used in a ReferenceArray Input.                                                         |
```
[the remark this answers is not in the document]
Comment (fzaninotto) [reply docs/AutocompleteArrayInput.md:58]: That's on purpose, to let the component name wrap and not force the table layout to go beyond the content width

File: docs/AutocompleteArrayInput.md [hunk 5]
```diff
@@ -10,125 +10,418 @@ It renders using MUI [Autocomplete](https://mui.com/components/autocomplete/).
...
+| `optionText`               | Optional | `string` &#124; `Function` &#124; `Component` | `name` | Field name of record to display in the suggestion item or function which accepts the correct record as argument (`(record)=> {string}`)           |
+| `optionValue`              | Optional | `string`              | `id`                     | Field name of record containing the value to use as input value                                                                                         |
+| `setFilter`                | Optional | `Function`            | `null`                   | A callback to inform the `searchText` has changed and new `choices` can be retrieved based on this `searchText`. Signature `searchText => void`. This function is automatically set up when using `ReferenceArray Input`. |
+| `shouldRender Suggestions` | Optional | `Function`            | `() => true`             | A function that returns a `boolean` to determine whether or not suggestions are rendered.  |
```
[the remark this answers is not in the document]
Comment (fzaninotto) [reply docs/AutocompleteArrayInput.md:68]: Same

File: docs/AutocompleteArrayInput.md [hunk 6]
```diff
@@ -10,125 +10,418 @@ It renders using MUI [Autocomplete](https://mui.com/components/autocomplete/).
...
+| `onCreate`                 | Optional | `Function`            | `-`                      | A function called with the current filter value when users choose to create a new choice.                                                               |
+| `optionText`               | Optional | `string` &#124; `Function` &#124; `Component` | `name` | Field name of record to display in the suggestion item or function which accepts the correct record as argument (`(record)=> {string}`)           |
+| `optionValue`              | Optional | `string`              | `id`                     | Field name of record containing the value to use as input value                                                                                         |
+| `setFilter`                | Optional | `Function`            | `null`                   | A callback to inform the `searchText` has changed and new `choices` can be retrieved based on this `searchText`. Signature `searchText => void`. This function is automatically set up when using `ReferenceArray Input`. |
```
[the remark this answers is not in the document]
Comment (fzaninotto) [reply docs/AutocompleteArrayInput.md:67]: same

**Pull request merged**